### PR TITLE
Maintenance/stdlib

### PR DESCRIPTION
**Pull request by christianparpart**
### CHECKLIST BEFORE MERGE

- [x] resolve review items
- [ ] have dependency git tags updates to their newly released versions

### How:

```sh
# fish shell syntax

for file in (git grep fmt/format | cut -d: -f1); sed -i -e 's,include <fmt/format.h>,include <format>,' $file; end
for file in (git grep fmt:: | cut -d: -f1); sed -i -e 's,fmt::,std::,g' $file; end
for file in (git grep std::print | cut -d: -f1); sed -i -e 's,std::print,std::cout << std::format,g' $file; end

# replace `std::print` with `std::cout << std::format`

## tidy-up
find src \( -name '*.cpp' -o -name '*.h' \) -exec clang-format -i {} \;

```

## misc notes

- check statusline's time in `%H:%S` is working correctly
- some lines wrt. `fmt::runtime()` have been rewritten
- `fmt::arg()` rewritten for `std::format()`
- Also fix timestamps in debug logs

Closes https://github.com/contour-terminal/contour/issues/1598

### CODE REVIEW

File: examples/watch-mouse-events.cpp [hunk 1]
```diff
@@ -225,7 +225,7 @@ struct MouseTracker final: public BasicParserEvents
 
         auto const state = _decrpm.value().second;
         auto const supported = state == 1 || state == 2;
-        fmt::print("Passive mouse tracking: {}\n", supported ? "supported" : "not supported");
+        std::cout << std::format("Passive mouse tracking: {}\n", supported ? "supported" : "not supported");
```
Comment (Yaraslaut) [inline examples/watch-mouse-events.cpp:228]: If we switch to c++23 we can call `std::print`

File: scripts/install-deps.sh [hunk 2]
```diff
@@ -110,18 +110,24 @@ fetch_and_unpack_termbenchpro()
 
 fetch_and_unpack_boxed()
 {
-    local boxed_cpp_version="1.4.2"
+    local boxed_cpp_git_sha="45e7c2e393dd1644f1ab18fb48ba59910c75aad0"
```
Comment (Yaraslaut) [inline scripts/install-deps.sh:113]: We also have a cmake variable for minimal boxed-cpp version `BOXED_CPP_MINIMAL_VERSION` in `cmake/ContourThirdParties.cmake` file

File: src/contour/Actions.h [hunk 3]
```diff
@@ -304,7 +303,7 @@ inline auto getDocumentation()
 // {{{ fmtlib custom formatters
 #define DECLARE_ACTION_FMT(T)                                                                          \
```
Comment (Yaraslaut) [inline src/contour/Actions.h:304]: we need to change name for this macro

File: src/contour/Config.cpp [hunk 4]
```diff
@@ -155,6 +155,11 @@ namespace
 #endif
     }
 
+    std::string replaceCommentPlaceholder(std::string const& docString)
+    {
+        return std::regex_replace(docString, std::regex { "\\{comment\\}" }, "#");
+    }
+
```
Comment (Yaraslaut) [inline src/contour/Config.cpp:162]: We might want to customize it for different writers in the future, so we should make it writer type dependent, something like 
```suggestion
    template <typename Writer=YAMLConfigWriter>
    std::string replaceCommentPlaceholder(std::string const& docString)
    {
        return std::regex_replace(docString, std::regex { "\\{comment\\}" }, Writer::comment);
    }

```

or make it a function of Writer type

File: src/contour/Config.h [hunk 5]
```diff
@@ -936,13 +935,15 @@ struct Writer
...
     {
-        return fmt::format("{}", v);
+        return std::format("{}", v);
     }
 
     template <typename... T>
     [[nodiscard]] std::string format(std::string_view doc, T... args)
     {
-        return fmt::format(fmt::runtime(doc), format(args)..., fmt::arg("comment", "#"));
+        std::string replacedDoc =
```
Comment (Yaraslaut) [inline src/contour/Config.h:944]: i think we need to use `replaceCommentPlaceholder` function here 
Also, why `vformat`?

File: src/crispy/CLI.cpp [hunk 6]
```diff
@@ -816,9 +816,9 @@ namespace // {{{ helpers
                               help_element::HelpText);
 
                 // {{{ append default value, if any
-                // NB: It seems like fmt::format is having problems with
+                // NB: It seems like std::format is having problems with
                 //     formatting std::variant<>'s on some systems.'
-                // auto const defaultValueStr = fmt::format("{}", option.value);
+                // auto const defaultValueStr = std::format("{}", option.value);
```
Comment (Yaraslaut) [inline src/crispy/CLI.cpp:821]: Can we delete this comment?

File: src/vtbackend/StatusLineBuilder.cpp [hunk 7]
```diff
@@ -305,7 +304,16 @@ struct VTSerializer
     std::string visit(StatusLineDefinitions::Clock const&)
     {
         crispy::ignore_unused(this);
-        return fmt::format("{:%H:%M}", fmt::localtime(std::time(nullptr)));
+
+        // NB: The following is printing the time in UTC, but we need it in local time.
```
Comment (Yaraslaut) [inline src/vtbackend/StatusLineBuilder.cpp:308]: Maybe add `TODO` so we have a chance to return here in the future

**Review by Yaraslaut — COMMENTED** [review]
Few small things to consider, 
I think that we can switch in this PR to c++23 and use std::print

File: src/contour/Config.cpp (hunk 4 above)
Comment (christianparpart) [reply src/contour/Config.cpp]: I think I'll leave potential config API/implementation improvements to you :)

File: src/contour/Config.h (hunk 5 above)
Comment (christianparpart) [reply src/contour/Config.h]: I tried also without `vformat`, but then the formatting string needs to be a literal (i.e. `std::format_string<Args...>`)

Comment (Yaraslaut) [reply src/contour/Config.h]: Oh, I was hoping that they made it into c++23, but only in c++26 https://en.cppreference.com/w/cpp/utility/format/runtime_format

File: scripts/ci-install-run-deps.sh
Comment (Yaraslaut) [inline scripts/ci-install-run-deps.sh:1]: we can use install-deps.sh to get all dependencies, why this file?

**Review by Yaraslaut — APPROVED** [review]
Many thanks, I think we can merge it
just one file in scripts not sure that we need it

**Comment by christianparpart** [comment]
Which file? I am in the forest right now. It's hard to check on the phone :)

**Comment by Yaraslaut** [comment]
> Which file? I am in the forest right now. It's hard to check on the phone :)

[scripts/ci-install-run-deps.sh](https://github.com/contour-terminal/contour/pull/1600/files/87bde6b2aa460fe0996f795a24f0e1fb5e1ae427#diff-54a0f34bae984b37737703e9a3775783e925bf718099ec8e83adf15c790face8) :)

File: scripts/ci-install-run-deps.sh
Comment (christianparpart) [reply scripts/ci-install-run-deps.sh:1]: we call it twice. I'd prefer to maintain the actual dependencies to install in as few locations as ossible :)

**Pull request merged**